onnxruntime/core/providers/webnn/builders/impl/attention_op_builder.cc

```diff
@@ -0,0 +1,568 @@
+
+  options.set("label", node.Name() + "range_of_mask_shape");
+  options.set("exclusive", true);
+  options.set("reversed", false);
+  emscripten::val neq_left = model_builder.GetBuilder().call<emscripten::val>(
+      "cumulativeSum", mask_shape_ones_shape_constant, gsl::narrow<uint32_t>(3), options);
+
+  std::vector<uint32_t> reshape_pre_neq_right = {past_sequence_length, qkv_sequence_length};
+  std::vector<int32_t> pre_neq_right_data_range(qkv_sequence_length);
+  std::iota(pre_neq_right_data_range.begin(), pre_neq_right_data_range.end(), 1);
+
+  emscripten::val desc_pre_neq_right_data_range = emscripten::val::object();
+  ORT_RETURN_IF_NOT(SetWebnnDataType(desc_pre_neq_right_data_range, ONNX_NAMESPACE::TensorProto_DataType_INT32),
+                    "Unsupported data type");
+  emscripten::val dims_pre_neq_right_data_range = emscripten::val::array(std::vector<uint32_t>({qkv_sequence_length}));
+  desc_pre_neq_right_data_range.set("dimensions", dims_pre_neq_right_data_range);
+  desc_pre_neq_right_data_range.set("shape", dims_pre_neq_right_data_range);
+  emscripten::val pre_neq_right_data_range_buffer =
+      emscripten::val::global("Int32Array").new_(emscripten::val::array(pre_neq_right_data_range));
+  emscripten::val pre_neq_right_data_range_constant = model_builder.GetBuilder().call<emscripten::val>(
+      "constant", desc_pre_neq_right_data_range, pre_neq_right_data_range_buffer);
+
+  common_options.set("label", node.Name() + "/GQA/attn_mask/add");
+  emscripten::val pre_neq_right = model_builder.GetBuilder().call<emscripten::val>(
+      "add", pre_neq_right_data_range_constant, scatter_pos, common_options);
+
+  common_options.set("label", node.Name() + "/GQA/expand_neq_right");
+  emscripten::val expanded_neq_right = model_builder.GetBuilder().call<emscripten::val>(
+      "expand", pre_neq_right, emscripten::val::array(reshape_pre_neq_right), common_options);
+
+  options.set("permutation", emscripten::val::array(std::vector<uint32_t>({1, 0})));
+  options.set("label", node.Name() + "/GQA/neq_right/transpose");
+  emscripten::val neq_right =
+      model_builder.GetBuilder().call<emscripten::val>("transpose", expanded_neq_right, options);
+
+  common_options.set("label", node.Name() + "/GQA/attn_mask/condition");
+  emscripten::val condition =
+      model_builder.GetBuilder().call<emscripten::val>("lesser", neq_left, neq_right, common_options);
+
+  std::vector<float_t> value_one({1});
+  emscripten::val desc_value_one = emscripten::val::object();
+  ORT_RETURN_IF_NOT(SetWebnnDataType(desc_value_one, ONNX_NAMESPACE::TensorProto_DataType_FLOAT),
+                    "Unsupported data type");
+  emscripten::val dims_one = emscripten::val::array(std::vector<uint32_t>({1}));
+  desc_value_one.set("dimensions", dims_one);
+  desc_value_one.set("shape", dims_one);
+  emscripten::val value_one_buffer = emscripten::val::global("Float32Array").new_(emscripten::val::array(value_one));
+  emscripten::val value_one_constant =
```

Comment:
We have a common util `CreateOrGetConstant` in `model_builder.h` used to get or create webnn constant with same value.